react/src/components/BookDetailCommentaries/index.js

```diff
@@ -1,28 +1,38 @@
 import React, { Component, Fragment } from 'react';
+import {connect} from 'react-redux';
 
 import CommentaryList from '../CommentaryList';
-import {data} from '../../mock.js';
+import BookService from '../../redux/books/actions';
 
 import CommentaryAdd from './CommentaryAdd';
 import strings from './strings';
 import './styles.css';
 
-class BookDetailCommentaries extends Component {
+const MAX_COMMENTARIES = 4;
 
+class BookDetailCommentaries extends Component {
+  componentWillMount() {
+    this.props.dispatch(BookService.getCommentaries(this.props.bookId));
+  }
   render() {
     const bookId = this.props.bookId;
-    const book = data.find(item => Number(item.id) === Number(bookId));
 
     return (
       <Fragment>
         <h2 className="green-subtitle">{strings.commentaries}</h2>
         <div className="commentaries-detail">
-          <CommentaryAdd />
-          {book && <CommentaryList commentaries={book.commentaries} />}
+          <CommentaryAdd bookId={this.props.bookId}/>
+          {bookId && <CommentaryList commentaries={this.props.currentBookCommentaries.sort((a, b) => b.id-a.id).slice(0, MAX_COMMENTARIES)} />}
```

Comment:
I'd use `reselect` to calculate this
```
this.props.currentBookCommentaries.sort((a, b) => b.id-a.id).slice(0, MAX_COMMENTARIES)
```
But I'd keep as a nice to have; optional.

Reply:
Done :smiley: